ocs_ci/ocs/ui/block_pool.py

```diff
@@ -0,0 +1,136 @@
+from selenium.webdriver.common.by import By
+from ocs_ci.helpers.helpers import create_unique_resource_name
+from ocs_ci.ocs.exceptions import PoolStateIsUnknow
+import ocs_ci.ocs.resources.pod as pod
+
+logger = logging.getLogger(__name__)
+
+
+class BlockPoolUI(PageNavigator):
+    """
+    User Interface Selenium for Block Pools page
+
+    """
+
+    def __init__(self, driver):
+        super().__init__(driver)
+        ocp_version = get_ocp_version()
+        self.bp_loc = locators[ocp_version]["block_pool"]
+        self.sc_loc = locators[ocp_version]["storageclass"]
+
+    def create_pool(self, replica, compression):
+        """
+        Create block pool via UI
+        Args:
+            replica (int): replica size usually 2,3
+            compression (bool): True to enable compression otherwise False
+        Return:
+            array: pool name (str) pool status (bool) #pool can be created with failure status
+
+        """
+        pool_name = create_unique_resource_name("test", "rbd-pool")
+        self.navigate_block_pool_page()
+        self.do_click(self.bp_loc["create_block_pool"])
+        self.do_send_keys(self.bp_loc["new_pool_name"], pool_name)
+        self.do_click(self.bp_loc["first_select_replica"])
+        if replica == 2:
+            self.do_click(self.bp_loc["second_select_replica_2"])
+        else:
+            self.do_click(self.bp_loc["second_select_replica_3"])
+        if compression is True:
+            self.do_click(self.bp_loc["conpression_checkbox"])
+        self.do_click(self.bp_loc["pool_confirm_create"])
+        wait_for_text_result = self.wait_for_element_text(self.bp_loc["pool_state_inside_pool"], "Ready",timeout=15)
+        if wait_for_text_result is True:
+            logger.info(f"Pool {pool_name} was created and it is in Ready state")
+            return [pool_name, True]
+        else:
+            logger.info(f"Pool {pool_name} was created but did not reach Ready state")
+            return [pool_name, False]
+
+    def check_pool_existence(self, pool_name):
+        """
+        Check if pool appears in the block pool list
+        Args:
+            pool_name (str): Name of the pool to check
+        Return:
+            bool: True if pool is in the list of pools page, otherwise False
+        """
+        self.navigate_overview_page()
+        self.navigate_block_pool_page()
+        self.page_has_loaded()
+        pool_existence = self.check_element_text(expected_text=pool_name)
+        logger.info(f"Pool name {pool_name} existence is {pool_existence}")
+        return pool_existence
+
+    def delete_pool(self, pool_name):
+        """
+        Delete pool from pool page
+        Args:
+            pool_name (str): The name of the pool to be deleted
+        Return:
+            bool: True if pool is not found in pool list, otherwise false
+        """
+        self.navigate_overview_page()
+        self.navigate_block_pool_page()
+        self.page_has_loaded()
```

Comment:
Is it mandatory to use this function? Does it not work without this function? we use 'WebDriverWait'.
All tests use the "navigate" function

Reply:
WebDriverWait is good when you search for an element in the next page. This is good for knowing that the page was loaded without searching for a specific element.